sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventHubsEventSource.cs

```diff
@@ -171,20 +171,24 @@ protected EventHubsEventSource() : base(EventSourceName)
         /// <param name="operationId">An artificial identifier for the publishing operation.</param>
         /// <param name="retryCount">The number of retries that were used for service communication.</param>
         /// <param name="eventCount">The number of events that were received in the batch.</param>
+        /// <param name="startingSequenceNumber">The sequence number of the first event in the batch.</param>
+        /// <param name="endingSequenceNumber">The sequence number of the last event in the batch.</param>
         /// <param name="durationSeconds">The total duration that the receive operation took to complete, in seconds.</param>
         ///
-        [Event(7, Level = EventLevel.Informational, Message = "Completed receiving events for Event Hub: {0} (Consumer Group: '{1}', Partition Id: '{2}'); Operation Id: '{3}'.  Service Retry Count: {4}; Event Count: {5}; Duration: '{6:0.00}' seconds")]
+        [Event(7, Level = EventLevel.Informational, Message = "Completed receiving events for Event Hub: {0} (Consumer Group: '{1}', Partition Id: '{2}'); Operation Id: '{3}'.  Service Retry Count: {4}; Event Count: {5}; Starting sequence number: '{6}', Ending sequence number: '{7}'; Duration: '{8:0.00}' seconds")]
         public virtual void EventReceiveComplete(string eventHubName,
                                                  string consumerGroup,
                                                  string partitionId,
                                                  string operationId,
                                                  int retryCount,
                                                  int eventCount,
+                                                 string startingSequenceNumber,
```

Comment:
Same question about putting new parameters in the middle - is this safe?

Reply:
Chatted offline.  To ensure that we don't break structured use, the original ordering to the `WriteEvent` call was restored and the new parameters appended.    Log message was left in the desired format, unfortunately requiring out-of-order slugs.